src/Components/Components/test/ComponentBaseTest.cs

```diff
@@ -261,6 +263,100 @@ public async Task DoesNotRenderAfterOnInitAsyncTaskIsCancelled()
             Assert.Equal(2, renderer.Batches.Count);
         }
 
+        [Fact]
+        public async Task RunsOnAfterRender_AfterRenderingCompletes()
+        {
+            // Arrange
+            var renderer = new TestRenderer();
+            var component = new TestComponent() { Counter = 1 };
+
+            var onAfterRenderCompleted = false;
+            component.OnAfterRenderLogic = (c, firstTime) =>
+            {
+                Assert.True(firstTime);
+                Assert.Single(renderer.Batches);
+                onAfterRenderCompleted = true;
+            };
+
+            // Act
+            var componentId = renderer.AssignRootComponentId(component);
+            var renderTask = renderer.RenderRootComponentAsync(componentId);
+
+            // Assert
+            await renderTask;
+            Assert.True(onAfterRenderCompleted);
+            Assert.Single(renderer.Batches);
+
+            // Component should not be rendered again. OnAfterRender doesn't do that.
+            Assert.Single(renderer.Batches);
```

Comment:
dupe of line 288 (suspect that line should be removed since the comment sounds reasonable)